QModManager/HarmonyPatches/MaskTelemetryLogging.cs

```diff
@@ -0,0 +1,189 @@
+                0x00BA : stelem.ref
+                0x00BB : dup
+                0x00BC : ldc.i4.1
+                0x00BD : ldloc.1
+                0x00BE : ldfld	System.Int32 sessionId
+                0x00BF : box	System.Int32
+                0x00C0 : stelem.ref
+                0x00C1 : call	Void LogFormat(System.String, System.Object[])
+                0x00C2 : ldc.i4.0
+                0x00C3 : ret
+                */
+                #endregion
+
+                //determinde the check to find the right place and not missmatch it accidentally with a other one
+                //v1
+                //if (codes[i].opcode == OpCodes.Ldfld && codes[i + 6].opcode == OpCodes.Box && codes[i +10].opcode == OpCodes.Ret)
+                //v2
+                if (codes[i].opcode == OpCodes.Ldstr && codes[i + 17].opcode == OpCodes.Box && codes[i + 21].opcode == OpCodes.Ret)
+                {
+                    Logger.Log(Logger.Level.Debug, $"{thistranspiler} - Found IL Code Line for Index. Index = {i.ToString()}");
+                    found = true;
+                    Index = i;
+                    break;
+                }
+            }
+
+            if (deeplogging)
+            {
+                if (found)
+                {
+                    Logger.Log(Logger.Level.Debug, $"{thistranspiler} - found: true");
+                }
+                else
+                {
+                    Logger.Log(Logger.Level.Debug, $"{thistranspiler} - found: false");
+                }
+            }
+
+            if (Index > -1)
+            {
+                Logger.Log(Logger.Level.Debug, $"{thistranspiler} - Transpiler injectection position found");
+                //v1
+                /*
+                codes[Index] = new CodeInstruction(OpCodes.Ldstr, "Masked by QModManager for privacy Reason");
+                codes.RemoveRange(Index, 0);
+                */
+                //v2
+                codes[Index] = new CodeInstruction(OpCodes.Ldstr, "Telemetry session started. Platform: '{0}', UserId: ->Masked by QModManager for privacy Reason<-, SessionId: {1}");
+                codes.RemoveRange(Index, 0);
+                codes.RemoveRange(Index+10, 5);
+            }
+            else
+            {
+                Logger.Log(Logger.Level.Error, $"{thistranspiler} - Index was not found");
+            }
+
+            //logging after
+            if (deeplogging)
+            {
+                Logger.Log(Logger.Level.Debug, $"{thistranspiler} - Deep Logging after-transpiler:");
+                for (int k = 0; k < codes.Count; k++)
+                {
+                    Logger.Log(Logger.Level.Debug, (string.Format("0x{0:X4}", k) + $" : {codes[k].opcode.ToString()}	{(codes[k].operand != null ? codes[k].operand.ToString() : "")}"));
+                }
+            }
+
+            Logger.Log(Logger.Level.Debug, $"{thistranspiler} - Transpiler end going to return");
+            return codes.AsEnumerable();
```

Comment:
Lists are enumerables, so this line can be changed to:
```suggestion
            return codes;
```